### PR TITLE
feat: Add interface index (#11)

### PR DESCRIPTION
**Pull request by YuiYukihira**
Hey there, 👋

I'm writing my own project that needs network interfaces, and this crate looks pretty good for my needs.

My only problem is that I need the interface index as well. I saw that there's a currently open issue (#11) for this but no PR,
so I thought I'd give it a go.

#### Testing
- [x] Works on MacOS
- [x] Works on Linux
- [ ] Haven't tested Windows, but I think it should work fine...

Let me know if there's any changes you think I should make.

### CODE REVIEW

**Comment by EstebanBorai** [comment]
Hi @YuiYukihira!

Thanks so much for opening this PR, it looks like the changes are not compiling for Windows.
Looks related to not passing the new `index` param to mostly.

Otherwise looks great!

**Comment by YuiYukihira** [comment]
Hey @EstebanBorai,

I've added a new commit to resolve the windows issues,
I also had to add a new error type because `winapi` uses `u32` sized errors codes, instead of `i32`.

Let me know before you merge anything and I'll squash this into a single commit.

File: src/target/windows.rs [hunk 1]
```diff
@@ -257,6 +266,20 @@ fn make_ipv6_netmask(_sockaddr: &*mut SOCKADDR_IN6) -> Netmask<Ipv6Addr> {
     None
 }
 
+fn get_adapter_address_index(adapter_address: &*mut AdapterAddress) -> Result<u32> {
+    let adapter_luid = &unsafe { (*(*adapter_address)).Luid } as *const IF_LUID;
+
+    let index = &mut 0 as *mut u32;
```
Comment (EstebanBorai) [inline src/target/windows.rs:272]: @YuiYukihira looks like this casting is not invalid.

**Comment by YuiYukihira** [comment]
Sorry about that, should all be fine this time.

**Comment by EstebanBorai** [comment]
> Sorry about that, should all be fine this time.

No worries!

**Review by EstebanBorai — APPROVED** [review]
LGTM!

**Comment by YuiYukihira** [comment]
Great, thanks!

Btw, do you have an idea when you might make a new release that incorporates this?

**Pull request merged**